frontend/packages/topology/src/components/application-panel/TopologyApplicationPanel.tsx

```diff
@@ -1,32 +1,42 @@
 import * as React from 'react';
+import { Node } from '@patternfly/react-topology';
 import { ResourceIcon, ActionsMenu } from '@console/internal/components/utils';
+import { ActionServiceProvider } from '@console/shared';
 import { groupActions } from '../../actions/groupActions';
-import { GraphData, TopologyApplicationObject } from '../../topology-types';
+import { TopologyApplicationObject } from '../../topology-types';
 import TopologyApplicationResources from './TopologyApplicationResources';
 
 type TopologyApplicationPanelProps = {
-  graphData: GraphData;
+  element: Node;
   application: TopologyApplicationObject;
 };
 
 const TopologyApplicationPanel: React.FC<TopologyApplicationPanelProps> = ({
-  graphData,
+  element,
   application,
-}) => (
-  <div className="overview__sidebar-pane resource-overview">
-    <div className="overview__sidebar-pane-head resource-overview__heading">
-      <h1 className="co-m-pane__heading">
-        <div className="co-m-pane__name co-resource-item">
-          <ResourceIcon className="co-m-resource-icon--lg" kind="application" />
-          {application.name}
-        </div>
-        <div className="co-actions">
-          <ActionsMenu actions={groupActions(graphData, application)} />
-        </div>
-      </h1>
+}) => {
+  const graphData = element.getGraph().getData();
+
+  return (
+    <div className="overview__sidebar-pane resource-overview">
+      <div className="overview__sidebar-pane-head resource-overview__heading">
+        <h1 className="co-m-pane__heading">
+          <div className="co-m-pane__name co-resource-item">
+            <ResourceIcon className="co-m-resource-icon--lg" kind="application" />
+            {application.name}
+          </div>
+          <div className="co-actions">
+            <ActionServiceProvider context={{ 'topology-context-actions': element }}>
+              {({ options, loaded }) =>
+                loaded && <ActionsMenu actions={groupActions(options, graphData, application)} />
```

Comment:
we shouldn't be calling groupActions here to get the delete action. It should come from 
ActionServiceProvider